Snake/BusinessLayer/Engine.cs

```diff
@@ -1,217 +1,217 @@
+
+        private enum gameMode
+        {
+            basic,
+        }
+        public enum snakeAction
+        {
+            eat,
+            die,
+            move
+        }
+
+        gameMode currentMode = gameMode.basic;
+
+        public Engine(int length, int width, int mode)
+        {
+            mazeLength = length;
+            mazeWidth = width;
+            Score.resetScore();
+            foodGenerator = new FoodGenerator();
+        }
+
+        public int[,] initializeGame()
+        {
+
+            switch (currentMode)
+            {
+                case gameMode.basic:
+
+
+                    // Create a New Maze and initialize it
+                    gameMaze = new Maze(mazeWidth, mazeLength);
+                    mazeArray = gameMaze.CreateMaze();
+
+                    // Add the Snake
+                    gameSnake1 = new GameSnake();
+                    //List<Point> snakeBody = new List<Point>();
+                    List<Point> snakeCurrentBody = gameSnake1.createFirstSnake(mazeLength, mazeWidth, snakeInitialLength);
+
+                    // Make the whole snake as body first
+                    foreach (Point value in snakeCurrentBody)
+                    {
+                        mazeArray[value.returnX(), value.returnY()] = (int)Elements.snakeBody;
+                    }
+                    // Identify snake head
+                    Point head = snakeCurrentBody[0];
+                    mazeArray[head.returnX(), head.returnY()] = (int)Elements.snakeHead;
+
+                    // Add the Food
+                    bool isValid = true;
+                    do
+                    {
+                        food = foodGenerator.generateFood(mazeLength, mazeWidth);
+                        isValid = validateNewFoodLocation(food);
+                    } while (!isValid);
+                    mazeArray[food.xLocation, food.yLocation] = Food;
+                    break;
+
+                default:
+                    throw new System.Exception("Invalid Game Mode!");
+            }
+
+            return mazeArray;
+        }
+
+        public int[,] updateGame(Direction snakeDirection)
+        {
+            if (snakeDirection == Direction.Unchanged)
+            {
+                snakeDirection = gameSnake1.directionFacing;
+            }
+
+            Point newSnakeHead = getNewHead(snakeDirection);
+            List<Point> snakesNewLocation;
+
+            switch (mazeArray[newSnakeHead.returnX(), newSnakeHead.returnY()])
+            {
+
+                case (int)Elements.mazeBody: 
+                    if (Score.getScore() > Score.getHighScore())
+                    {
+                        Score.setHighScore(Score.getScore());
+                    }
+                    mazeArray[0, 0] = snakeHitsMaze;
+                    return mazeArray;
+
+
+                case Food:  // snake hits the food
+                    snakesNewLocation = gameSnake1.snakeMove(snakeDirection, true);
```

Comment:
this is a big chunk of code of a case. consider if some of this should be in their own methods with single responsibilities